pkg/controller/channel/channel_controller.go

```diff
@@ -217,125 +239,119 @@ func (r *ReconcileChannel) Reconcile(request reconcile.Request) (reconcile.Resul
-			localLabels := secrectInstance.GetLabels()
-			if localLabels == nil {
-				localLabels = make(map[string]string)
-			}
-
-			localLabels[chv1.ServingChannel] = "true"
-			secrectInstance.SetLabels(localLabels)
+	//sync the channel to the serving-channel annotation in all involved secrets.
+	srtRef := instance.Spec.SecretRef
 
-			err = r.Update(context.TODO(), secrectInstance)
-			klog.Infof("Set label serving-channel to secret object: %#v, error: %#v", *secrectInstance, err)
-		}
-		//sync the channel to the serving-channel annotation in all involved secrets.
-		r.syncSecrectAnnotation(instance, request.NamespacedName)
+	if err := r.updatedReferredObjLabel(srtRef, srtGvk); err != nil {
+		klog.Errorf("failed to update referred secret label %v", err)
 	}
 
-	r.updateConfigMap(instance)
+	if err := r.syncReferredObjAnnotation(request, srtRef, srtGvk); err != nil {
+		klog.Errorf("faild to annotation %v", err)
+	}
 
 	//sync the channel to the serving-channel annotation in all involved ConfigMaps.
-	r.syncConfigAnnotation(instance, request.NamespacedName)
+	//r.syncConfigAnnotation(instance, request.NamespacedName)
+	cmRef := instance.Spec.ConfigMapRef
+	if err := r.updatedReferredObjLabel(cmRef, cmGvk); err != nil {
+		klog.Errorf("failed to update referred configMap label %v", err)
+	}
+
+	if err := r.syncReferredObjAnnotation(request, cmRef, cmGvk); err != nil {
+		klog.Errorf("faild to annotation %v", err)
+	}
 
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileChannel) updateConfigMap(instance *chv1.Channel) {
-	if instance.Spec.ConfigMapRef != nil && instance.Spec.ConfigMapRef.Name > "" {
-		configName := instance.Spec.ConfigMapRef.Name
-		configNamespace := instance.Spec.ConfigMapRef.Namespace
+func (r *ReconcileChannel) updatedReferredObjLabel(ref *corev1.ObjectReference, objGvk schema.GroupVersionKind) error {
+	if ref == nil {
+		return gerr.New(fmt.Sprintf("empty referred object %v", objGvk.Kind))
+	}
 
-		if configNamespace == "" {
-			configNamespace = instance.Namespace
-		}
+	objName := ref.Name
+	objNs := ref.Namespace
 
-		configInstance := &corev1.ConfigMap{}
-		configKey := types.NamespacedName{Name: configName, Namespace: configNamespace}
+	obj := &unstructured.Unstructured{}
+	objKey := types.NamespacedName{Name: objName, Namespace: objNs}
 
-		err := r.Get(context.TODO(), configKey, configInstance)
-		if err == nil {
-			localLabels := configInstance.GetLabels()
+	obj.SetGroupVersionKind(objGvk)
 
-			if localLabels == nil {
-				localLabels = make(map[string]string)
-			}
+	if err := r.Get(context.TODO(), objKey, obj); err != nil {
+		return gerr.Wrapf(err, "failed to get the referred object %v", objGvk.Kind)
+	}
 
-			localLabels[chv1.ServingChannel] = "true"
-			configInstance.SetLabels(localLabels)
+	localLabels := obj.GetLabels()
+	if localLabels == nil {
+		localLabels = make(map[string]string)
+	}
 
-			err = r.Update(context.TODO(), configInstance)
+	localLabels[chv1.ServingChannel] = "true"
```

Comment:
Sounds like we are using chv1.ServingChannel both as an annotation and as a label name? ...

Reply:
yes, for the label, we are using it just simply mark the referenced object. 

whenever a cluster reconcile comes, we use the label to select the referenced object to update annotation of them, adding/deleting cluster-info to the annotation. This annotation is consumed by other teams.